### PR TITLE
Merge book into repo

### LINKED ISSUE #309: Move book into main repo
We'll never keep the book up to date if it's not in the main repo and built on CI.

So we should:

* [ ] move the book from https://github.com/graphql-rust/graphql-rust.github.io into this repo under /book
* [ ] fix up the compile tests of the book, which are not running properly ATM
* [ ] extend the CI to build the book tests
* [ ] update the book where appropriate
* [ ] extend the release scripts to bump version numbers in the book

### PR DESCRIPTION
**Pull request by theduke**
Closes #309.

A few points:

* I switched to mdbook to avoid the node dependency
* This adds a unrelated feature in the first commit (specifying context for #[derive(GraphQLObject)]).
  I was too lazy to split it out.
* Book now uses 2018 edition in all examples
* Adds a CI job for testing the examples
* I did not merge the commits from the books repo, let me know if you feel otherwise here

Todo: update links (since the links will now be https://graphql-rust.github.io/juniper) and figure out how we want to handle updating the book.

Should it be done automatically on each merge?
In that case the CI could build the book after a merge and auto-commit & push to a "book" or "gh-pages" branch.

We could also have two version of the book, the currently released version and in a subdirectory the master, with a link at the top to go to the master.
So we would update the master version on each merge, but the regular one only on release.

thoughts @LegNeato ?

### CODE REVIEW

**Comment by LegNeato** [comment]
Awesome, thanks! Yeah, I think we should do it automatically and I think we should have two versions.

File: docs/book/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Work in progress with improved documentation and main website for Juniper.
+
+## Requirements
```
Comment (LegNeato) [inline docs/book/README.md]: Remove node stuff

File: docs/book/content/README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+[iron]: http://ironframework.io
+[graphql_spec]: http://facebook.github.io/graphql
+[test_schema_rs]: https://github.com/graphql-rust/juniper/blob/master/juniper/src/tests/schema.rs
+[tokio]: https://github.com/tokio-rs/tokio
+[hyper_examples]: https://github.com/graphql-rust/juniper/tree/master/juniper_hyper/examples
+[rocket_examples]: https://github.com/graphql-rust/juniper/tree/master/juniper_rocket/examples
+[iron_examples]: https://github.com/graphql-rust/juniper/tree/master/juniper_iron/examples
+[hyper]: https://hyper.rs
+[rocket]: https://rocket.rs
+[book]: https://graphql-rust.github.io
```
Comment (LegNeato) [inline docs/book/content/README.md]: Update this.

File: docs/book/.travis.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+language: rust
```
Comment (LegNeato) [inline docs/book/.travis.yml]: This whole file should be deleted, right?

File: docs/book/README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+- Node.js
+- `yarn`
+
+## Contributing
```
Comment (LegNeato) [inline docs/book/README.md]: These aren't right anymore.

**Comment by theduke** [comment]
@LegNeato you need to give the "GraphQL Rust Team" admin permissions on Azure.

I've created a variable group with the ssh keys for pushing from Azure but I can't link it to the project due to missing permissions.

**Comment by LegNeato** [comment]
@theduke I think I made the team an admin now, hopefully that works

**Comment by LegNeato** [comment]
@theduke have you tried to get the pushing to work after I tweaked permissions?

**Comment by theduke** [comment]
@LegNeato yeah.

Script works locally and will work on CI *IF* the ssh task works fine, but we will only see that once we merge.

**Comment by theduke** [comment]
Appveyor is driving me crazy with it's incomplete docs...

I'm merging this for now without CI actually working.
I'll try to figure out the CI tomorrow.

**Pull request merged**